Ensures that `children` prop is not required by Button component (#509)
mongodb/leafygreen-ui: .changeset/short-vans-unite.md

---
'@leafygreen-ui/button': patch
---

> Ensures that `children` is not a required prop on component

Reviewer: ```suggestion
Fix type signature to not have `children` be a required prop
```

Author: Accidentally merged too soon but will make a branch shortly to fix this
